app/models/metric/purging.rb

```diff
@@ -50,45 +50,34 @@ def self.purge_timer(ts, interval)
     end
   end
 
-  def self.purge_count(older_than, interval)
-    klass, conditions = case interval
-                        when 'realtime' then [Metric,       {}]
-                        else             [MetricRollup, {:capture_interval_name => interval}]
-                        end
-
-    oldest = klass.select(:timestamp).where(conditions).order(:timestamp).first
-    oldest = oldest.nil? ? older_than : oldest.timestamp
+  def self.klass_for_interval(interval)
+    case interval
+    when 'realtime' then [Metric, Metric]
+    else                 [MetricRollup, MetricRollup.where(:capture_interval_name => interval)]
```

Comment:
:sparkles: object, but you might be able to simplify this by just returning the scope, and pulling the class out of it later...  something like

```ruby
def self.scope_for_interval(interval)
  case interval
  when 'realtime' then Metric.all
  else                 MetricRollup.where(:capture_interval_name => interval)
  end
end
```

Then later on

```ruby
scope = scope_for_interval(interval)
klass = scope.klass
```